components/SearchSandbox/index.js

```diff
@@ -157,9 +160,12 @@ class SearchSandbox extends Component {
 	}
 
 	toggleAnalytics = () => {
-		this.setState(prevState => ({
-			analytics: !prevState.analytics,
-		}));
+		this.setState((prevState) => {
+			this.pref.analytics = !prevState.analytics;
```

Comment:
Instead of updating context over here. We can instead update it in `savePreferences` using state.